XYZDrivers/src/main/java/com/xyzdrivers/repositories/PaymentsRepo.java

```diff
@@ -18,38 +18,61 @@
      * Retrieve ALL data from MEMBERS table
      *
      * @return List of MembershipPayments objects
+     * @throws com.xyzdrivers.repositories.RepositoryException
      */
     @Override
     public List<MembershipPayment> get() throws RepositoryException {
-        DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
-        DateFormat timeFormat = new SimpleDateFormat("HH:mm:ss");
         List<Object[]> results;
-        List<MembershipPayment> payments = new ArrayList();
 
         try {
-            //get data
+            // Retrieve raw database data, then expand into MembershipPayment objects
             results = this.sqlService.retrieve("MEMBERS");
-            //parse members data
-            for (Object[] memberData : results) {
-                Calendar date = Calendar.getInstance();
-                Calendar time = Calendar.getInstance();
-                date.setTime(dateFormat.parse(memberData[3].toString()));
-                time.setTime(timeFormat.parse(memberData[4].toString()));
-
-                MembershipPayment payment = new MembershipPayment(memberData[0].toString(), // mem_id
-                        memberData[1].toString(), // Type of payment
-                        (float) memberData[2],
-                        date, // date
-                        time); // time
-                payments.add(payment);
-            }
-        } catch (SQLException | ParseException ex) {
+            return translateMembershipPayments(results);
+        } catch (SQLException ex) {
             throw new RepositoryException("Failed to retrieve payments. See inner exception for details.", ex);
         }
+    }
+
+    private List<MembershipPayment> translateMembershipPayments(List<Object[]> flattenedPayments) throws RepositoryException {
+        List<MembershipPayment> payments = new ArrayList<>();
+
+        for (Object[] memberData : flattenedPayments) {
+            Calendar dateTime = combineDateTime(memberData[3], memberData[4]);
+
+            MembershipPayment payment = new MembershipPayment(memberData[0].toString(), // mem_id
+                    memberData[1].toString(), // Type of payment
+                    (float) memberData[2],
```

Comment:
I think when I was fixing some bugs I saw you can't cast to a float like this even though netbeans doesnt complain

Reply:
:(